frontend/src/components/ScenarioComponents/CardsComponents/GroupFilter/FilterCard.tsx

```diff
@@ -0,0 +1,128 @@
+// SPDX-FileCopyrightText: 2024 German Aerospace Center (DLR)
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import {Box, useTheme} from '@mui/material';
+import Divider from '@mui/material/Divider';
+import FilterRows from './FilterRows';
+import CardTitle from '../MainCard/CardTitle';
+
+import {Localization} from 'types/localization';
+import {Dictionary} from 'util/util';
+
+interface FilterCardProps {
+  /** Title of the filter card */
+  title: string;
+
+  /** Color of the filter card */
+  color: string;
+
+  /** Array of compartments */
+  compartments: string[];
+
+  /** Dictionary of filtered values */
+  filteredValues: Dictionary<number> | null;
+
+  /** Index of the group filter */
+  groupFilterIndex: number;
+
+  /** Total number of cards */
+  totalCardNumber: number;
+
+  /** Boolean to determine if the compartment is expanded */
+  compartmentExpanded?: boolean;
+
+  /** Selected compartment */
+  selectedCompartment: string;
+
+  /** Minimum number of compartment rows */
+  minCompartmentsRows: number;
+
+  /** Maximum number of compartment rows */
+  maxCompartmentsRows: number;
+
+  /** A boolean indicating whether the compartments are expanded. */
+  compartmentsExpanded: boolean;
+
+  /** An object containing localization information (translation & number formattation).*/
+  localization?: Localization;
+}
+
+/**
+ * This component renders a filter card with a title, and a list of values that are the filtered values.
+ * It also supports localization.
+ */
+export default function FilterCard({
+  title,
+  color,
+  compartments,
+  filteredValues,
+  groupFilterIndex,
+  totalCardNumber,
+  compartmentExpanded,
+  selectedCompartment,
+  maxCompartmentsRows,
+  minCompartmentsRows,
+  compartmentsExpanded,
+  localization = {
+    formatNumber: (value: number) => value.toString(),
+    customLang: 'global',
+    overrides: {},
+  },
+}: FilterCardProps) {
+  const theme = useTheme();
+  return (
+    <Box
+      id={`external-cardfilter-container-${groupFilterIndex}`}
+      sx={{
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'center',
+      }}
+    >
+      <Box
+        id={`cardfilter-title&cardContent-container-${groupFilterIndex}`}
+        className='hide-scrollbar'
+        sx={{
+          maxHeight: compartmentsExpanded
+            ? maxCompartmentsRows > 5
+              ? `${(390 / 6) * maxCompartmentsRows}px`
+              : `${(480 / 6) * maxCompartmentsRows}px`
+            : minCompartmentsRows < 4
+              ? `${(365 / 4) * minCompartmentsRows}px`
+              : `${(325 / 4) * minCompartmentsRows}px`,
```

Comment:
Nested ternary operators are discouraged. Refactor this out to a `useMemo` with `if/else`.